addons/sale_stock/models/sale_order.py

```diff
@@ -153,17 +148,16 @@ def _compute_picking_ids(self):
         for order in self:
             order.delivery_count = len(order.picking_ids)
 
-    @api.onchange('company_id')
-    def _onchange_company_id(self):
-        if self.company_id:
-            warehouse_id = self.env['ir.default'].get_model_defaults('sale.order').get('warehouse_id')
-            self.warehouse_id = warehouse_id or self.user_id.with_company(self.company_id.id)._get_default_warehouse_id().id
-
-    # @api.onchange('user_id')
-    # def onchange_user_id(self):
-    #     super().onchange_user_id()
-    #     if self.state in ['draft', 'sent']:
-    #         self.warehouse_id = self.user_id.with_company(self.company_id.id)._get_default_warehouse_id().id
+    @api.depends('user_id', 'company_id')
+    def _compute_warehouse_id(self):
+        default_warehouse_id = self.env['ir.default'].get_model_defaults('sale.order').get('warehouse_id')
```

Comment:
defaults are company dependent, should probably be done in the loop, with a with_company :thinking: